py/usageaccountant/datadog_fetcher.py

```diff
@@ -0,0 +1,307 @@
+                   "family":"bytes",
+                   "id":2,
+                   "name":"byte",
+                   "short_name":"B",
+                   "plural":"bytes",
+                   "scale_factor":1.0
+                },
+                None
+             ],
+             "query_index":0,
+             "aggr":"avg",
+             "metric":"redis.mem.peak",
+             "tag_set":["shared_resource_id: rc_long_redis"],
+             "expression":"avg: redis.mem.peak{shared_resource_id:
+             rc_long_redis,app_feature: shared}.rollup(5)",
+             "scope":"app_feature: shared, shared_resource_id: rc_long_redis",
+             "interval":5,
+             "length":2,
+             "start":1721083885000,
+             "end":1721083894000,
+             "pointlist":[
+                [1721083885000.0, 2494386918.6666665],
+                [1721083890000.0, 2494386918.6666665]
+             ],
+             "display_name":"redis.mem.peak",
+             "attributes":{}
+          }
+       ],
+       "values":[],
+       "times":[],
+       "message":"",
+       "group_by":["shared_resource_id"]
+    }
+
+    series_list: List of series dict extracted from API response
+    parsed_unit: UsageUnit parsed from API response
+    usage_accumulated: Object of UsageAccumulator class
+    """
+    for series in series_list:
+        scope = series.get("scope", "")
+        scope_dict = parse_response_scope(scope)
+        if (
+            "shared_resource_id" not in scope_dict.keys()
+            or "app_feature" not in scope_dict.keys()
+        ):
+            exception_msg = (
+                "Required parameters, shared_resource_id and/or app_feature "
+                "not found in series.scope of the series received."
+            )
+            raise InvalidSeriesError(exception_msg, series)
+
+        resource = scope_dict["shared_resource_id"]
+        app_feature = scope_dict["app_feature"]
+        for point in series.get("pointlist", []):
+            # TODO remove this before prod
+            # print(f"resource: {resource}, app_feature: {app_feature}, "
+            #       f"amount: {int(point[1])}, usage_type: {parsed_unit}")
+            usage_accumulator.record(
+                resource_id=resource,
+                app_feature=app_feature,
+                # point[0] is the timestamp
+                amount=int(point[1]),
+                usage_type=parsed_unit,
+            )
+
+
+def main(
+    query_file: TextIO,
+    start_time: int,
+    end_time: int,
+    kafka_config_file: TextIO,
+) -> None:
+    """
+    query: File with Datadog queries, one per line
+    start_time: Start of the time window in Unix epoch format
+                with second-level precision
+    end_time: End of the time window in Unix epoch format
+              with second-level precision
+    kafka_config: File with parameters to initialize UsageAccumulator object
+    """
+    kafka_config = json.loads(kafka_config_file.read())
```

Comment:
Please validate the content of this file. Ensure the mandatory fields are present at least.